rs/bitcoin/adapter/src/get_successors_handler.rs

```diff
@@ -97,7 +97,7 @@ impl GetSuccessorsHandler {
             .observe(request.processed_block_hashes.len() as f64);
 
         let response = {
-            let state = self.state.lock().await;
+            let state = self.state.lock().unwrap();
```

Comment:
Similarly, get_successors doesn't need to be async, so then all tests don't need to await on it / be async.

Reply:
This one I left on purpose because I am still unsure what to do with the channels i.e. blockchain_manager_tx